added BMKG as a new source

chunshek opened the pull request:
I have added Badan Meteorologi, Klimatologi, dan Geofisika (BMKG) as a new official source for Indonesia. I have implemented the following:

- Current Observations
- Hourly Forecasts
- Warnings
  - Nowcasting warnings (short-term, in Indonesian only)
  - Impact Based Forecasts (full day, in both Indonesian and English)

This source also provides reverse geolocation.

papjul commented on app/src/src_nonfreenet/org/breezyweather/sources/bmkg/BmkgService.kt:
```diff
@@ -0,0 +1,286 @@
+    companion object {
+        private const val BMKG_BASE_URL = "https://cuaca.bmkg.go.id/"
+        // The API key can be found from tokenDefault in the HTML source of https://cuaca.bmkg.go.id/
+        private const val BMKG_API_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjFjNWFkZWUxYzY5MzM0NjY2N2EzZWM0MWRlMjBmZWZhNDcxOTNjYzcyZDgwMGRiN2ZmZmFlMWVhYjcxZGYyYjQiLCJpYXQiOjE3MDE1ODMzNzl9.D1VNpMoTUVFOUuQW0y2vSjttZwj0sKBX33KyrkaRMcQ"
```
Needs to be implemented as a build config `BMKG_KEY`. `.github/workflows/push.yml` needs an update to add the key there.

chunshek replied:
Fixed.

papjul commented on app/src/src_nonfreenet/org/breezyweather/sources/bmkg/BmkgService.kt:
```diff
@@ -0,0 +1,286 @@
+        ignoreFeatures: List<SecondaryWeatherSourceFeature>
+    ): Observable<WeatherWrapper> {
+        // API Key is needed for warnings, but not for current/forecast.
+        val apiKey = BMKG_API_KEY
```
Make it a preference so that the user can change it if it suddenly stops working

chunshek replied:
Fixed.

chunshek commented:
Updated the following:
- moved API key to use build config `BMKG_KEY` / `breezy.bmkg.key`
- added `ConfigurableSource` for user supplied API key
- moved API key in queries from `@GET` parameters to `@HeaderMap`
- added `FEATURE_AIR_QUALITY` with current measurement for PM2.5 for both main and secondary

papjul commented:
Cleanup commit: https://github.com/breezy-weather/breezy-weather/commit/0c3dd57b7bd423a0990324d1524490d3a43ffd00

If Hong Kong can access other weather sources with no issues (firewall), then we can suggest other sources in the location preset. Wdyt?

chunshek commented:
Continuing this conversation in #1417
